pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -555,58 +551,16 @@ func (reconciler *Reconciler) setConsoleHost(cr *api.KieApp, env api.Environment
 		consoleCN = cr.Spec.CommonConfig.ApplicationName
 		cr.Status.ConsoleHost = fmt.Sprintf("http://%s", consoleCN)
 	}
-	reconciler.reconcileConsoleLink(cr, consoleLink)
 	return consoleCN
 }
 
-func (reconciler *Reconciler) reconcileConsoleLink(cr *api.KieApp, expected *consolev1.ConsoleLink) {
-	current := &consolev1.ConsoleLink{}
-	if cr.Status.ConsoleLink != "" {
-		err := reconciler.Service.Get(context.TODO(), types.NamespacedName{Name: cr.Status.ConsoleLink}, current)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				cr.Status.ConsoleLink = ""
-			} else {
-				log.Warn("Unable to retrieve existing ConsoleLink: ", cr.Status.ConsoleLink, err)
-			}
-			return
-		}
-	}
-	var err error
-	if expected == nil {
-		if cr.Status.ConsoleLink != "" {
-			log.Debug("Delete consoleLink")
-			err = reconciler.Service.Delete(context.TODO(), current)
-			if err == nil {
-				cr.Status.ConsoleLink = ""
-			}
-		}
-	} else {
-		if cr.Status.ConsoleLink == "" {
-			log.Debug("Create consoleLink for ", cr.Name)
-			err = reconciler.Service.Create(context.TODO(), expected)
-			if err == nil {
-				cr.Status.ConsoleLink = expected.Name
-				controllerutil.AddFinalizer(cr, constants.ConsoleLinkFinalizer)
-			}
-		} else if current.Spec.Href != expected.Spec.Href {
-			log.Debug("Update consoleLink with: ", expected.Spec.Href)
-			current.Spec.Href = expected.Spec.Href
-			_, err = reconciler.UpdateObj(current)
-		}
-	}
-	if err != nil {
-		log.Warn("Unable to reconcile ConsoleLink. ", err)
-	}
-}
-
 func (reconciler *Reconciler) getKubernetesResources(cr *api.KieApp, env api.Environment) []resource.KubernetesResource {
 	var resources []resource.KubernetesResource
 	objects := filterOmittedObjects(getCustomObjects(env))
 	for _, obj := range objects {
 		resources = append(resources, reconciler.getCustomObjectResources(obj, cr)...)
 	}
-	return resources
+	return reconciler.addConsoleLinkResources(resources, cr)
```

Comment:
Change the function to a `get`, and return `nil` if one is not to be added. Do the append here. Passing on the array makes code hard to follow and error-prone.